client/packages/reports/src/components/ReportWidget.tsx

```diff
@@ -39,17 +39,19 @@ export const ReportWidget: React.FC<PropsWithChildren<ReportWidgetProps>> = ({
   setReportWithArgs,
 }) => {
   const navigate = useNavigate();
-
-  const { setId, setArgs } = useReportStore();
+  const { updateQuery } = useUrlQuery();
 
   const reportArgs = useCallback(
     (report: ReportRowFragment, args: JsonData | undefined) => {
-      setArgs(report.id, args);
+      const stringifyArgs = JSON.stringify(args);
+      updateQuery({ reportArgs: stringifyArgs });
       navigate(
-        RouteBuilder.create(AppRoute.Reports).addPart(report.id).build()
+        RouteBuilder.create(AppRoute.Reports)
+          .addPart(`${report.id}?reportArgs=${stringifyArgs}`)
```

Comment:
Not quite happy with this... 🤔 probably a better way to redirect than hardcoding 👀